solidity/contracts/deposit/DepositFunding.sol

```diff
@@ -55,7 +55,7 @@ library DepositFunding {
         uint256 _n,
         uint256 _lotSizeSatoshis
     ) public returns (bool) {
-        TBTCSystem _system = TBTCSystem(_d.TBTCSystem);
+        TBTCSystem _system = TBTCSystem(address(_d.tbtcSystem));
```

Comment:
Why can't we just reference `_d.tbtcSystem` here? It's already properly typed, no?

Reply:
_d.tbtcSystem is an `ITBTCSystem` instance. It doesn't have functions like `getAllowNewDeposits()`

Reply:
It's worth adding necessary functions to the interface actually. In this PR if that's okay with you @Shadowfiend

Reply:
Yes please, awesome plan!